src/code/z_sram_NES.c

```diff
@@ -35,70 +35,90 @@ PersistentCycleSceneFlags sPersistentCycleSceneFlags[SCENE_MAX] = {
+//!       Only the mask is read from the `PERSISTENT_` macros.
 u16 sPersistentCycleWeekEventRegs[ARRAY_COUNT(gSaveContext.save.saveInfo.weekEventReg)] = {
-    /*  0 */ WEEK_FLG(WEEKEVENTREG_ENTERED_TERMINA_FIELD) | WEEK_FLG(WEEKEVENTREG_ENTERED_IKANA_GRAVEYARD) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_ROMANI_RANCH) | WEEK_FLG(WEEKEVENTREG_ENTERED_GORMAN_TRACK) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_MOUNTAIN_VILLAGE_WINTER) | WEEK_FLG(WEEKEVENTREG_ENTERED_GORON_SHRINE) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_SNOWHEAD),
-    /*  1 */ WEEK_FLG(WEEKEVENTREG_ENTERED_SOUTHERN_SWAMP_POISONED) | WEEK_FLG(WEEKEVENTREG_ENTERED_WOODFALL) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_DEKU_PALACE) | WEEK_FLG(WEEKEVENTREG_ENTERED_GREAT_BAY_COAST) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_PIRATES_FORTRESS) | WEEK_FLG(WEEKEVENTREG_ENTERED_ZORA_HALL) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_WATERFALL_RAPIDS) | WEEK_FLG(WEEKEVENTREG_ENTERED_IKANA_CANYON),
-    /*  2 */ WEEK_FLG(WEEKEVENTREG_ENTERED_IKANA_CASTLE) | WEEK_FLG(WEEKEVENTREG_ENTERED_STONE_TOWER) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_STONE_TOWER_INVERTED) | WEEK_FLG(WEEKEVENTREG_ENTERED_EAST_CLOCK_TOWN) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_WEST_CLOCK_TOWN) | WEEK_FLG(WEEKEVENTREG_ENTERED_NORTH_CLOCK_TOWN) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_WOODFALL_TEMPLE) | WEEK_FLG(WEEKEVENTREG_ENTERED_SNOWHEAD_TEMPLE),
-    /*  3 */ WEEK_FLG(WEEKEVENTREG_ENTERED_PIRATES_FORTRESS_EXTERIOR) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_STONE_TOWER_TEMPLE) | WEEK_FLG(WEEKEVENTREG_ENTERED_STONE_TOWER_TEMPLE_INVERTED) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_THE_MOON) | WEEK_FLG(WEEKEVENTREG_ENTERED_MOON_DEKU_TRIAL) |
-        WEEK_FLG(WEEKEVENTREG_ENTERED_MOON_GORON_TRIAL) | WEEK_FLG(WEEKEVENTREG_ENTERED_MOON_ZORA_TRIAL) |
-        WEEK_FLG(WEEKEVENTREG_03_80),
+    /*  0 */ PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_TERMINA_FIELD) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_IKANA_GRAVEYARD) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_ROMANI_RANCH) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_GORMAN_TRACK) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_MOUNTAIN_VILLAGE_WINTER) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_GORON_SHRINE) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_SNOWHEAD),
+    /*  1 */ PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_SOUTHERN_SWAMP_POISONED) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_WOODFALL) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_DEKU_PALACE) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_GREAT_BAY_COAST) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_PIRATES_FORTRESS) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_ZORA_HALL) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_WATERFALL_RAPIDS) |
+        PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_IKANA_CANYON),
+    /*  2 */ PERSISTENT_WEEKEVENTREG(WEEKEVENTREG_ENTERED_IKANA_CASTLE) |
```

Comment:
Probably not a bad idea to put all these comments on their own lines when there's several flags, you can see that some format that way by themselves and some do not. Realistically the whole array's probably going to end up as one line per flag anyway.

Reply:
Since there's so many 0 entries, I only put it on the line above when there's 3 or more flags, not sure if that's alright